modules/mesh-task/main.tf

```diff
@@ -207,6 +207,70 @@ resource "aws_ecs_task_definition" "this" {
     name = local.consul_binary_volume_name
   }
 
+  dynamic "volume" {
```

Comment:
If there is a better way to do this, let me know.

It's translating a list of maps (`volumes = [{...}]`) into several blocks (`volume {}`). From what I can tell, Terraform doesn't support passing blocks into modules. And, it doesn't support dynamic names for the keys in these dynamic blocks.

Reply:
Eek, this doesn't look fun. Did this come from somewhere else? If so, it would be great to link to the source.

Reply:
> Did this come from somewhere else?

No, I crafted this myself. I followed the [Terraform docs on dynamic blocks](https://www.terraform.io/docs/language/expressions/dynamic-blocks.html) and referred to the [volumes field in the `aws_ecs_task_definition`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_task_definition#volume) and looked at a some examples of nested dynamic blocks.

(One way we maybe might avoid this is to implement mesh-task as a Terraform provider, but that's essentially a rewrite that we'd need to think through.)